spec/wibox/test_utils.lua

```diff
@@ -22,7 +22,9 @@ end
 local function stub_draw_widget(wibox, cr, widget, x, y, width, height)
     assert.is.equal("wibox", wibox)
     assert.is.equal("cr", cr)
-    table.insert(widgets_drawn, { widget, x, y, width, height })
+    if widget.visible then
+        table.insert(widgets_drawn, { widget, x, y, width, height })
+    end
```

Comment:
I'd just skip this chunk. Surely tests involving invisible widgets would still like to know if such a widget was (tried to) drawn.